vectorizers/_vectorizers.py

```diff
@@ -74,10 +76,22 @@ def construct_token_dictionary_and_frequency(token_sequence, token_dictionary=No
     return token_dictionary, token_frequency, n_tokens
 
 
+def select_tokens_by_regex(tokens, regex):
+    if not isinstance(regex, re.Pattern):
+        regex = re.compile(regex)
+    result = set([])
+    for token in tokens:
+        if regex.match(token) is not None:
```

Comment:
Do you re.match or re.fullmatch? MWETransformer uses fullmatch at the moment.

Reply:
I switched to fullmatch. Makes sense.